packages/auth/src/grant/routes.ts

```diff
@@ -140,6 +140,9 @@ async function createApprovedGrant(
     await trx.commit()
   } catch (err) {
     await trx.rollback()
+    if (typeof err === 'string') {
+      throw new GNAPServerRouteError(400, GNAPErrorCode.InvalidRequest, err)
+    }
```

Comment:
I think it would be useful to map errors from the services to the HTTP status & GNAPErrorCodes, similar to how we are doing in the backend's incoming payments for example.